Chess+LibSQL: Remove IODevice, improve error propagation

BenWiederhake opened the pull request:
This migrates Chess and ChessEngine from IODevice to Core::File (and removes an unused include from LibSQL).

Advances #17129, so CC @AtkinsSJ 

Note that now IODevice seems to be only used by DeprecatedFile and Ladybird (looks like an unused include, but I'm not sure).

GMTA commented on Userland/Services/ChessEngine/ChessEngine.h:
```diff
@@ -26,8 +33,8 @@ class ChessEngine : public Chess::UCI::Endpoint {
 private:
-    ChessEngine(NonnullOwnPtr<Core::File> in, NonnullOwnPtr<Core::File> out)
-        : Endpoint(move(in), move(out))
+    explicit ChessEngine()
```
Why is this made `explicit`? I think we only do that for single-argument constructors. We don't really seem to care about initializer lists, for example.

BenWiederhake replied:
The history behind this is that for a short few seconds, `ChessEngine` was indeed a single-arg constructor in my IDE, and then I changed it, and apparently forgot to remove the "explicit". :)

GMTA commented on Userland/Games/Chess/Engine.cpp:
```diff
@@ -54,13 +54,11 @@ void Engine::connect_to_engine_service()
-    auto infile = Core::DeprecatedFile::construct();
-    infile->open(rpipefds[0], Core::OpenMode::ReadOnly, Core::DeprecatedFile::ShouldCloseFileDescriptor::Yes);
-    set_in(infile);
+    auto infile = Core::File::adopt_fd(rpipefds[0], Core::File::OpenMode::Read, Core::File::ShouldCloseFileDescriptor::Yes).release_value_but_fixme_should_propagate_errors();
```
As discussed on Discord, the `ShouldCloseFileDescriptor` argument can be left to its default if you'd like.

BenWiederhake commented:
Changes since initial push:
- Dropped the default-value to open (`Core::File::ShouldCloseFileDescriptor::Yes`), thanks to @fdellwing and @GMTA
- Un-explicited `ChessEngine::ChessEngine()`, thanks to @GMTA
- Rebased for convenience
